### PR TITLE
feat!: new quic-rpc, simlified generics, bump MSRV to 1.76

### PR DESCRIPTION
**Pull request by Frando**
## Description

Adapts iroh for n0-computer/quic-rpc#76

Wit this change, a quic client is
`pub type QuicIroh = Iroh<QuinnConnection<RpcService>>`
instead of
`pub type QuicIroh = Iroh<QuinnConnection<ProviderRequest, ProviderResponse>>`

## Breaking Changes
* Increased mimimal supported rust version (MSRV) from 1.75 to 1.76
* The generics of the iroh clients changed: Instead of taking `ProviderRequest, ProviderResponse` as generics, they now take `RpcService`

## Change checklist

- [x] Self-review.
- [x] Documentation updates if relevant.
- [x] Tests if relevant.
- [x] All breaking changes documented.

### CODE REVIEW

**Comment by Frando** [comment]
Needs rebase

**Comment by Frando** [comment]
Rebased this. Needs a quic-rpc release first though.

**Comment by dignifiedquire** [comment]
release is done

**Comment by Frando** [comment]
Rebased again and bumped quic-rpc. Should be good to go now.

File: Cargo.lock [hunk 1]
```diff
@@ -1151,6 +1170,19 @@ dependencies = [
  "serde",
 ]
 
+[[package]]
+name = "derive_more"
+version = "0.99.17"
```
Comment (dignifiedquire) [inline Cargo.lock:1175]: huh, looks like quic-rpc has outdated deps -.-

File: Cargo.lock [hunk 2]
```diff
@@ -4043,16 +4075,19 @@ dependencies = [
...
 [[package]]
 name = "quic-rpc"
-version = "0.9.0"
+version = "0.10.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "d6f365b77f07281f6dfc300eb5a9913185fa686e9119c787200ace337a53bdbf"
+checksum = "3131faef53f1eae5d2f394b2ce6f130326245863f3f29b639dcc3f5616c88316"
 dependencies = [
  "bincode",
+ "blake3",
```
Comment (dignifiedquire) [inline Cargo.lock:4083]: hmm

**Review by dignifiedquire — APPROVED** [review]

File: Cargo.lock (hunk 2 above)
Comment (Frando) [reply Cargo.lock:4083]: Where does that come from..?

**Pull request merged**